client/analytics.ts

```diff
@@ -46,41 +64,71 @@ export default class Analytics {
     amplitude.getInstance().logEvent('Scan Parse Error')
   }
 
-  static scanCompleted({ timeTaken, successRatio }) {
+  static scanCompleted({
+    timeTaken,
+    successRatio,
+  }: {
+    timeTaken: number
+    successRatio: number
+  }) {
     amplitude.getInstance().logEvent('Scan Parse Completed', {
       successRatio,
       timeTaken,
     })
   }
 
-  static performedExportsAnalysis(packageName) {
+  static performedExportsAnalysis(packageName: string) {
     amplitude.getInstance().logEvent('Exports Analysis Performed', {
       package: packageName,
     })
   }
 
-  static exportsAnalysisSuccess({ packageName, timeTaken }) {
+  static exportsAnalysisSuccess({
+    packageName,
+    timeTaken,
+  }: {
+    packageName: string
+    timeTaken: number
+  }) {
     amplitude.getInstance().logEvent('Exports Analysis Successful', {
       package: packageName,
       timeTaken,
     })
   }
 
-  static exportsAnalysisFailure({ packageName, timeTaken }) {
+  static exportsAnalysisFailure({
+    packageName,
+    timeTaken,
+  }: {
+    packageName: string
+    timeTaken: number
+  }) {
     amplitude.getInstance().logEvent('Exports Analysis Failed', {
       package: packageName,
       timeTaken,
     })
   }
 
-  static exportsSizesSuccess({ packageName, timeTaken }) {
+  static exportsSizesSuccess({
+    packageName,
+    timeTaken,
+  }: {
+    packageName: string
+    timeTaken: number
+  }) {
     amplitude.getInstance().logEvent('Exports Size Calculated', {
       package: packageName,
       timeTaken,
     })
   }
 
-  static exportsSizesFailure({ packageName, timeTaken }) {
+  static exportsSizesFailure({
+    packageName,
+    timeTaken,
+  }: {
+    packageName: string
```

Comment:
You could avoid duplication by having a type like:

```
type HasPackageName = { packageName: string }
type HasTimeTaken = { timeTaken: number }
```

And then intersect those together and compose argument types on each method

Reply:
Good point. I'll revise.